File: server/users-microservice/src/api/signcheck/signCheckController.py
from flask_restful import Resource
from flask import jsonify, request

import jwt

from config.config import jwtSecret

class SignCheckController(Resource):
    
    def get(self):
        if 'jwt' in request.headers and 'csrf' in request.headers:
            token = request.headers['jwt']

            options = {
                'verify_exp': False,
                'verify_aud': False,
                'verify_iat': False
            }

            decoded_token = jwt.decode(token, key=jwtSecret, options=options)

            if decoded_token['csrf'] == request.headers['csrf']:
                response = jsonify(name=decoded_token['userName'], email=decoded_token['userEmail'])
                response.status_code = 200

                return response
            else:
                response = jsonify(message = 'The token is not valid')
                response.status_code = 401

                return response
        else:
            response = jsonify(message = 'Token and or Cookie missing')
            response.status_code = 401

            return response

File: server/users-microservice/src/api/signcheck/signcheckController.js
'use strict';

import { buildResponse } from '../../utils/responseService';
import { decodeToken } from '../../utils/tokenController';

export const save = (req, res) => {
  if (req.headers.jwt && req.headers.csrf) {
    let token = req.headers.jwt;
    let csrf = req.headers.csrf;

    let decodedToken = decodeToken(token);

    if (decodedToken.csrf == csrf) {
      let data = {
        name: decodedToken.userName,
        email: decodedToken.userEmail
      };

      buildResponse(200, data, res);
    } else {
      buildResponse(401, 'The token is not valid', res); 
    }
  } else {
    buildResponse(401, 'Token and/or Cookie missing', res);
  }
};